### PR TITLE
Finishing commit for f_39

### PR DESCRIPTION
**Pull request by tacticalDevC**
Finished f_39. Got a merge conflict because of #10

### CODE REVIEW

File: app/src/main/java/at/tacticaldevc/oat/utils/Cam.java [hunk 1]
```diff
@@ -1,7 +1,146 @@
...
+
+import at.tacticaldevc.oat.exceptions.OATApplicationException;
+
+import static at.tacticaldevc.oat.utils.Ensurer.ensureNotNull;
+
 /**
  * A helper class for the camera
  */
 public class Cam {
+    public static CameraCharacteristics CHARACTERISTICS;
```
Comment (FiryDragon1) [inline app/src/main/java/at/tacticaldevc/oat/utils/Cam.java]: Only final properties are usually written in Caps Lock.

File: app/src/main/java/at/tacticaldevc/oat/utils/Cam.java [hunk 2]
```diff
@@ -1,7 +1,156 @@
...
+        @Override
+        public void onConfigured(@NonNull CameraCaptureSession session) {
+            // Well guys we did it!
+            ccs = session;
+            green_light = true;
+        }
+
+        @Override
+        public void onConfigureFailed(@NonNull CameraCaptureSession session) {
+            // Seems like something's wrong
```
Comment (FiryDragon1) [inline app/src/main/java/at/tacticaldevc/oat/utils/Cam.java]: Shouldn't there be an error handler

Comment (tacticalDevC) [reply app/src/main/java/at/tacticaldevc/oat/utils/Cam.java]: True. Gimme a sec

File: app/src/main/java/at/tacticaldevc/oat/utils/Cam.java (hunk 1 above)
Comment (FiryDragon1) [reply app/src/main/java/at/tacticaldevc/oat/utils/Cam.java]: was fixed, resolving

File: app/src/main/java/at/tacticaldevc/oat/utils/Cam.java [hunk 3]
```diff
@@ -1,7 +1,156 @@
...
+            throw OATApplicationException.forLibraryError("Camera2", e);
+        }
+    }
+
+    /**
+     * If the camera has been opened and configured call this method to take a picture
+     * @return Image object to process further
+     */
+    public static Image takePicture() {
+        while (!green_light) {
```
Comment (FiryDragon1) [inline app/src/main/java/at/tacticaldevc/oat/utils/Cam.java]: endless loop

**Pull request merged**